base/range.jl

```diff
@@ -53,11 +53,16 @@ Given a starting value, construct a range either by length or from `start` to `s
 optionally with a given step (defaults to 1, a [`UnitRange`](@ref)).
 One of `length` or `stop` is required.  If `length`, `stop`, and `step` are all specified, they must agree.
 
-If `length` and `stop` are provided and `step` is not, the step size will be computed
-automatically such that there are `length` linearly spaced elements in the range (a [`LinRange`](@ref)).
+In most cases if `step` and `stop` are provided and `length` is not, the overall range length will be computed automatically such that the elements are `step` spaced (a [`StepRange`](@ref)).
```

Comment:
```suggestion
In most cases if `step` and `stop` are provided and `length` is not, the overall range length will
be computed automatically such that the elements are `step` spaced (a [`StepRange`](@ref)).
```

Reply:
The original documentation was describing the case with both `length` and `stop`, and we should continue to document that case:
```
If numeric `length` and `stop` values provided and `step` is not, the return value is a [`StepRangeLen`](@ref)
with a computed step.  However, for non-numeric values of `start` and `stop` (such as arrays), the `range` function instead returns a  [`LinRange`](@ref) object.  (Both `StepRangeLen` and `LinRange` correspond to equally spaced points up to roundoff errors, but deal with floating-point rounding slightly differently.)
```